lookout/style/typos/metrics.py

```diff
@@ -0,0 +1,96 @@
+import copy
+from enum import Enum, unique
+import os
+import pprint
+from typing import Dict, List, Set, Tuple
+
+import jinja2
+import pandas
+from sklearn.metrics import accuracy_score, precision_recall_fscore_support
+
+from lookout.style.typos.utils import Columns
+
+
+@unique
+class ScoreMode(Enum):
+    """Modes for calculation scores of typos correction."""
+
+    detection = "detection"
+    correction = "correction"
+    on_corrected = "on_corrected"
+
+
+def _load_jinja2_template(report_template_filename: str) -> jinja2.Template:
+    env = jinja2.Environment(trim_blocks=True, lstrip_blocks=True, keep_trailing_newline=True,
+                             extensions=["jinja2.ext.do"])
+    loader = jinja2.FileSystemLoader((os.path.join(os.path.dirname(__file__), "..", "templates"),),
+                                     followlinks=True)
+    template = loader.load(env, report_template_filename)
+    # the following is really needed, otherwise e.g. range is undefined
+    template.globals = template.environment.globals
+    return template
+
+
+def first_k_set(corrections: List[Tuple[str, float]], k: int) -> Set[str]:
+    """
+    Compose the set of k most probable correction candidates (tokens without probabilities).
+
+    :param corrections: List of corrections, sorted by probability.
+    :param k: Number of corrections to take.
+    :return: Set of k most probable correction tokens.
+    """
+    first_k = set()
+    for correction, _prob in corrections[:k]:
+        first_k.add(correction)
+    return first_k
```

Comment:
```python
return set(correction for correction, _ in corrections[:k])
```

Reply:
For completeness it's also possible to use

```python
    return {correction for correction, _ in corrections[:k]}
```
for even more syntax sugar or
```python
    return frozenset(correction for correction, _ in corrections[:k])
```
for even more immutability